File: GameLibrary/src/cc/lib/zombicide/ZTile.java
package cc.lib.zombicide;

import cc.lib.game.GRectangle;

public class ZTile {
    public final String id;
    public final int orientation;
    public GRectangle quadrant;

    public ZTile(String id, int orientation, GRectangle quadrant) {
        this.id = id;
        this.orientation = orientation;
        this.quadrant = quadrant;
    }

    public static GRectangle getQuadrant(int row, int col) {
        return new GRectangle(col, row, 3, 3);
    }

    public static GRectangle getQuadrant(int row, int col, int colsWidth, int rowsHeight) {
        return new GRectangle(col, row, colsWidth, rowsHeight);
    }

}

File: GameLibrary/src/cc/lib/zombicide/ZTile.kt
package cc.lib.zombicide

import cc.lib.game.GRectangle

class ZTile(@JvmField val id: String, @JvmField val orientation: Int,@JvmField  var quadrant: GRectangle) {
    companion object {
        @JvmStatic
        fun getQuadrant(row: Int, col: Int): GRectangle {
            return GRectangle(col.toFloat(), row.toFloat(), 3f, 3f)
        }

        @JvmStatic
        fun getQuadrant(row: Int, col: Int, colsWidth: Int, rowsHeight: Int): GRectangle {
            return GRectangle(col.toFloat(), row.toFloat(), colsWidth.toFloat(), rowsHeight.toFloat())
        }
    }
}